src/applistprovider.h

```diff
@@ -18,25 +18,36 @@ class AppListProvider : public QObject {
 
   virtual ~AppListProvider() = default;
 
+  // A Stable Identifier for an Application - should not change between
+  // invokations.
+  typedef QString AppId;
+  struct AppListEntry {
```

Comment:
As a minor suggestion, we could reuse the `AppPermission::AppDescription` class rather than defining a new type here.